pkg/utils/o11y/controller.go

```diff
@@ -44,24 +47,75 @@ func (h *SuiteController) convertBytesToMB(valueInBytes float64) int {
 	return int(valueInMegabytes)
 }
 
-// Fetch metrics for given query
-func (h *SuiteController) GetMetrics(query string) ([]MetricResult, error) {
+func (h *SuiteController) GetSecretName(namespace string, pattern string) (string, error) {
+	secrets, err := h.KubeInterface().CoreV1().Secrets(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, secret := range secrets.Items {
+		if strings.Contains(secret.Name, pattern) {
+			return secret.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("no matching secrets found")
+}
+
+func (h *SuiteController) GetDecodedTokenFromSecret(namespace, secretName string) (string, error) {
+	secret, err := h.KubeInterface().CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	tokenData, ok := secret.Data["token"]
+	if !ok {
+		return "", fmt.Errorf("token not found in secret %s", secretName)
+	}
+	Token := string(tokenData)
+
+	return Token, nil
+}
 
+func (h *SuiteController) QueryThanosAPI(namespace, thanosQuerierHost, token, query string) ([]MetricResult, error) {
 	var result struct {
 		Data struct {
 			Result []MetricResult `json:"result"`
 		} `json:"data"`
 	}
+	encodedQuery := url.QueryEscape(query)
+	url := fmt.Sprintf("https://%s/api/v1/query?query=%s", thanosQuerierHost, encodedQuery)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	// Custom HTTP client that ignores certificate errors
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
 
-	// Temporary way to fetch the metrics, will be replaced by golang http client library
-	// curl -X GET -kG "https://$THANOS_QUERIER_HOST/api/v1/query?" --data-urlencode "query="+query -H "Authorization: Bearer $TOKEN"
-	curlCmd := exec.Command("curl", "-X", "GET", "-kG", "http://localhost:8080/api/v1/query", "--data-urlencode", "query="+query)
-	output, err := curlCmd.Output()
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
There's a deprecation warning for `io/ioutils`. Can we use `io.ReadAll` here instead?